Hotfix/claim amt geo block css

W3stside opened the pull request:
Fixes:
1. Mobile size geoBlock warning of countries blocked placement
2. App breaking from incorrect ETH balance value
  a.  value was returning null and array value null was breaking
3. App breaking from wrong address passed to PriceFeedOracle due to async `.at` function

TODO:
- [x] URL redirect fix

W3stside committed:
paved over compile errors

W3stside committed:
geoBlock css fix

W3stside committed:
package update

W3stside committed:
fixed whitescreen issue from undefined method use

W3stside committed:
Fixed broken ETH Token (WETH) balance

W3stside committed:
fixed types

W3stside committed:
locked in versions compatible with latest dx-contracts DutchExchange version

W3stside committed:
update package-lock for broken ethereumjs-abi

1. hd-wallet-provider had broken dependency for ethereumjs-abi

W3stside committed:
removed p-lock to re-add after merge

W3stside committed:
merged in contract update 1.0.9

W3stside committed:
updated package-lock.json

W3stside committed:
updated package-lock.json pt.2

ghost commented:
Travis automatic deployment:
 https://pr438--dxreact.review.gnosisdev.com

ghost commented:
Travis automatic deployment:
 https://pr438--dxreact.review.gnosisdev.com

W3stside commented on src/api/index.ts:
```diff
@@ -129,7 +129,7 @@ export const getTokenBalances = async (tokenList: DefaultTokenObject[], account?
     decimals: token.decimals,
     address: token.address,
-    balance: balances[i] as BigNumber,
+    balance: balances[i] || toBigNumber(0) as BigNumber,
```
this fixes `null` `WETH` balance, though _why_ is `getETHBalance` returning `null` in the first place?

Velenir replied:
I would somewhat understand if it was `undefined`, but `null`... I'm stumped

W3stside replied:
me neither. but since this is happening with web3.eth.getBalance, sounds like not our issue. yay pointing fingers!

W3stside committed:
Changed deployedContracts.DutchExchange.ethUSDOracle

1. explicitly CALL fn

W3stside commented on src/api/contracts.ts:
```diff
@@ -173,10 +173,12 @@ async function init(provider: Provider) {
     delete deployedContracts.TokenOWLProxy
 
-    const oracleAddress = await deployedContracts.DutchExchange.ethUSDOracle()
+    const oracleAddress = await deployedContracts.DutchExchange.ethUSDOracle.call()
```
Change to explicitly call `ethUSDOracle` - @Velenir @anxolin could this have any say in the sometimes returned `0x` address? Truffle explicitly wraps non state changing fns with `.call` but maybe something there broke
/shrug

Velenir replied:
No, shouldn't be any different

W3stside replied:
will keep this way anyways but idk /shrug

ghost commented:
Travis automatic deployment:
 https://pr438--dxreact.review.gnosisdev.com

ghost commented:
Travis automatic deployment:
 https://pr438--dxreact.review.gnosisdev.com

Velenir commented on src/api/contracts.ts:
```diff
@@ -169,17 +169,19 @@ async function init(provider: Provider) {
 
     const { address: proxyAddress } = deployedContracts.DutchExchangeProxy
-    deployedContracts.DutchExchange = contractsMap.DutchExchange.at(proxyAddress)
+    deployedContracts.DutchExchange = await contractsMap.DutchExchange.at<Promise<DXAuction>>(proxyAddress)
```
What if instead of specifying `template param` you change `SimpleContract` in `api/types.ts` interface to return `): Promise<T>`

Velenir replied:
Also there's an instance of `HumanFriendly.at` in Tokens.ts

W3stside replied:
:+1:

W3stside replied:
after discussion w/me you anxo - this has been scrapped for synchronous `.at` calls

Velenir commented on src/containers/MenuAuctions/index.ts:
```diff
@@ -13,4 +13,4 @@ const mapStateToProps = (state: State) => ({
 })
 
-export default connect(mapStateToProps, { claimSellerFundsFromSeveral, push })(MenuAuctions)
+export default connect(mapStateToProps, { claimSellerFundsFromSeveral, push })(MenuAuctions as any)
```
TS gives me headaches

W3stside replied:
same :(

W3stside committed:
types fix - thanks dima

ghost commented:
Travis automatic deployment:
 https://pr438--dxreact.review.gnosisdev.com

ghost commented:
Travis automatic deployment:
 https://pr438--dxreact.review.gnosisdev.com

W3stside committed:
Revert/pin packages that worked on silent launch

1. Web3@0.20.7
2. dx-contract@1.0.5

W3stside committed:
contracts.ts PriceOracleInterface instantiation change

1. dont use dynamic DX PriceOracleInterface address
a. use networks.json address and deploy

W3stside committed:
revert TruffleContract.at type in api

W3stside committed:
fixed slow.trade/rinkeby URL redirect issue

W3stside committed:
fix feeRatio === NaN bug

W3stside committed:
pinned packages + lock

W3stside committed:
ignore broken hot reload type

W3stside committed:
return on successful App online check

W3stside committed:
MenuAuctions scss fix

ghost commented:
Travis automatic deployment:
 https://pr438--dxreact.review.gnosisdev.com

ghost commented:
Travis automatic deployment:
 https://pr438--dxreact.review.gnosisdev.com
